packages/framework/attributor/src/encoders.ts

```diff
@@ -108,6 +166,46 @@ export class AttributorSerializer implements IAttributorSerializer {
 			const user = JSON.parse(interner.getString(ref)) as IUser;
 			entries[i] = [key, { user, timestamp }];
 		}
+
+		if (customAttributionInfo !== undefined) {
+			const {
+				ids,
+				timestamps: encodedTimestamps2,
+				attributionRefs: attributionRefs2,
+				attributionPropsRefs,
+				attributionValuesRefs,
+			} = customAttributionInfo;
+			const timestamps2 = this.timestampEncoder.decode(encodedTimestamps2);
+			assert(
+				ids.length === timestamps2.length &&
+					ids.length === attributionRefs2.length &&
+					ids.length === attributionPropsRefs.length &&
+					ids.length === attributionValuesRefs.length,
+				"serialized cutom attribution columns should have the same length",
+			);
+			const entries2: [string, CustomAttributionInfo][] = Array.from({ length: ids.length });
+			for (const [i, key] of ids.entries()) {
+				// Non null asserting, we asserted ids, timestamps and attributionRefs have the same length above
+				// eslint-disable-next-line @typescript-eslint/no-non-null-assertion
+				const timestamp = timestamps2[i]!;
```

Comment:
Curious, why do you need to use `!` here and other places? They all seem to be plain typed arrays, and quickly checking in TS playground, it gives me actual type of the element, without adding undefined to it.